### PR TITLE
Refactor Storage

### PR DESCRIPTION
**Pull request by remo5000**
### Features
- Change `AddressBook` to `TaskManager` and `Person` to `Task` in `/storage` classes
  - Changes usage of these classes outside of `storage` as a consequence
- Modify XML files that are used for tests

### CODE REVIEW

File: src/main/java/seedu/address/storage/XmlAdaptedLabel.java [hunk 1]
```diff
@@ -8,35 +8,35 @@
 /**
  * JAXB-friendly adapted version of the Tag.
  */
-public class XmlAdaptedTag {
+public class XmlAdaptedLabel {
 
     @XmlValue
     private String tagName;
```
Comment (jiaxiu11) [inline src/main/java/seedu/address/storage/XmlAdaptedLabel.java]: refactor this into labelName?

File: src/main/java/seedu/address/storage/XmlAdaptedTask.java [hunk 2]
```diff
@@ -34,51 +35,51 @@
...
      * This is the no-arg constructor that is required by JAXB.
      */
-    public XmlAdaptedPerson() {}
+    public XmlAdaptedTask() {}
 
     /**
      * Constructs an {@code XmlAdaptedPerson} with the given person details.
      */
-    public XmlAdaptedPerson(String name, String phone, String email, String address, List<XmlAdaptedTag> tagged) {
+    public XmlAdaptedTask(String name, String phone, String email, String address, List<XmlAdaptedLabel> labelled) {
```
Comment (jiaxiu11) [inline src/main/java/seedu/address/storage/XmlAdaptedTask.java]: parameter names

File: src/test/java/seedu/address/storage/XmlAddressBookStorageTest.java [hunk 3]
```diff
@@ -110,8 +110,8 @@ public void saveAddressBook_nullAddressBook_throwsNullPointerException() {
      */
     private void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) {
```
Comment (jiaxiu11) [inline src/test/java/seedu/address/storage/XmlAddressBookStorageTest.java]: parameter name

File: src/test/java/seedu/address/storage/XmlSerializableAddressBookTest.java [hunk 4]
```diff
@@ -17,36 +17,36 @@
...
     @Test
-    public void toModelType_typicalPersonsFile_success() throws Exception {
-        XmlSerializableAddressBook dataFromFile = XmlUtil.getDataFromFile(TYPICAL_PERSONS_FILE,
-                XmlSerializableAddressBook.class);
-        AddressBook addressBookFromFile = dataFromFile.toModelType();
+    public void toModelType_typicalTasksFile_success() throws Exception {
+        XmlSerializableTaskManager dataFromFile = XmlUtil.getDataFromFile(TYPICAL_TASKS_FILE,
+                XmlSerializableTaskManager.class);
+        AddressBook taskManagerFromFile = dataFromFile.toModelType();
         AddressBook typicalPersonsAddressBook = TypicalPersons.getTypicalAddressBook();
```
Comment (jiaxiu11) [inline src/test/java/seedu/address/storage/XmlSerializableAddressBookTest.java]: local variable name

File: src/test/java/seedu/address/ui/PersonListPanelTest.java [hunk 5]
```diff
@@ -81,8 +81,8 @@ public void performanceTest() throws Exception {
      */
     private ObservableList<Person> createBackingList(int personCount) throws Exception {
         Path xmlFile = createXmlFileWithPersons(personCount);
-        XmlSerializableAddressBook xmlAddressBook =
-                XmlUtil.getDataFromFile(xmlFile, XmlSerializableAddressBook.class);
+        XmlSerializableTaskManager xmlAddressBook =
```
Comment (jiaxiu11) [inline src/test/java/seedu/address/ui/PersonListPanelTest.java]: local variable name

File: src/main/java/seedu/address/MainApp.java [hunk 6]
```diff
@@ -62,7 +62,7 @@ public void init() throws Exception {
 
         UserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(config.getUserPrefsFilePath());
         userPrefs = initPrefs(userPrefsStorage);
-        AddressBookStorage addressBookStorage = new XmlAddressBookStorage(userPrefs.getTaskManagerFilePath());
+        TaskManagerStorage addressBookStorage = new XmlTaskManagerStorage(userPrefs.getTaskManagerFilePath());
```
Comment (remo5000) [inline src/main/java/seedu/address/MainApp.java]: local variable

File: src/main/java/seedu/address/MainApp.java [hunk 7]
```diff
@@ -85,7 +85,7 @@ private Model initModelManager(Storage storage, UserPrefs userPrefs) {
         Optional<ReadOnlyTaskManager> addressBookOptional;
         ReadOnlyTaskManager initialData;
         try {
-            addressBookOptional = storage.readAddressBook();
+            addressBookOptional = storage.readTaskManager();
```
Comment (remo5000) [inline src/main/java/seedu/address/MainApp.java]: local variable

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 8]
```diff
@@ -21,11 +21,11 @@
 public class StorageManager extends ComponentManager implements Storage {
 
     private static final Logger logger = LogsCenter.getLogger(StorageManager.class);
-    private AddressBookStorage addressBookStorage;
+    private TaskManagerStorage addressBookStorage;
     private UserPrefsStorage userPrefsStorage;
 
 
-    public StorageManager(AddressBookStorage addressBookStorage, UserPrefsStorage userPrefsStorage) {
+    public StorageManager(TaskManagerStorage addressBookStorage, UserPrefsStorage userPrefsStorage) {
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/StorageManager.java]: local variable

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 9]
```diff
@@ -52,39 +52,39 @@ public void saveUserPrefs(UserPrefs userPrefs) throws IOException {
...
+    public Optional<ReadOnlyTaskManager> readTaskManager(Path filePath) throws DataConversionException, IOException {
         logger.fine("Attempting to read data from file: " + filePath);
-        return addressBookStorage.readAddressBook(filePath);
+        return addressBookStorage.readTaskManager(filePath);
     }
 
     @Override
-    public void saveAddressBook(ReadOnlyTaskManager addressBook) throws IOException {
-        saveAddressBook(addressBook, addressBookStorage.getAddressBookFilePath());
+    public void saveTaskManager(ReadOnlyTaskManager addressBook) throws IOException {
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/StorageManager.java]: params

File: src/main/java/seedu/address/storage/XmlAdaptedTask.java [hunk 10]
```diff
@@ -90,34 +90,34 @@ public Task toModelType() throws IllegalValueException {
...
+        if (description == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     Description.class.getSimpleName()));
         }
-        if (!Description.isValidDescription(address)) {
+        if (!Description.isValidDescription(description)) {
             throw new IllegalValueException(Description.MESSAGE_DESCRIPTION_CONSTRAINTS);
         }
-        final Description modelAddress = new Description(address);
+        final Description modelAddress = new Description(description);
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlAdaptedTask.java]: local variable

File: src/main/java/seedu/address/storage/XmlFileStorage.java [hunk 11]
```diff
@@ -27,10 +27,10 @@ public static void saveDataToFile(Path file, XmlSerializableAddressBook addressB
     /**
      * Returns address book in the file or an empty address book
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlFileStorage.java]: Comment

File: src/main/java/seedu/address/storage/XmlSerializableAddressBook.java [hunk 12]
```diff
@@ -21,22 +21,22 @@
...
 
     @XmlElement
-    private List<XmlAdaptedPerson> persons;
+    private List<XmlAdaptedTask> tasks;
 
     /**
      * Creates an empty XmlSerializableAddressBook.
      * This empty constructor is required for marshalling.
      */
     public XmlSerializableAddressBook() {
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlSerializableAddressBook.java]: - [ ] Rename after merging in UI

File: src/main/java/seedu/address/storage/XmlSerializableAddressBook.java [hunk 13]
```diff
@@ -47,12 +47,12 @@ public XmlSerializableAddressBook(ReadOnlyTaskManager src) {
      */
     public TaskManager toModelType() throws IllegalValueException {
         TaskManager addressBook = new TaskManager();
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlSerializableAddressBook.java]: local variable

File: src/main/java/seedu/address/storage/XmlTaskManagerStorage.java [hunk 14]
```diff
@@ -18,65 +18,63 @@
...
-     * Similar to {@link #readAddressBook()}
-     *
+     * Similar to {@link #readTaskManager()}
      * @param filePath location of the data. Cannot be null
      * @throws DataConversionException if the file is not in the correct format.
      */
-    public Optional<ReadOnlyTaskManager> readAddressBook(Path filePath) throws DataConversionException,
-            FileNotFoundException {
+    public Optional<ReadOnlyTaskManager> readTaskManager(Path filePath) throws DataConversionException,
+                                                                                 FileNotFoundException {
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlTaskManagerStorage.java]: formatting

File: src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml [hunk 15]
```diff
@@ -3,15 +3,15 @@
...
     <persons>
         <name>Hans Muster</name>
-        <phone isPrivate="false">948asdf2424</phone>
-        <email isPrivate="false">hans@example.com</email>
-        <address isPrivate="false">4th street</address>
+        <dueDate isPrivate="false">948asdf2424</dueDate>
+        <priorityValue isPrivate="false">hans@example.com</priorityValue>
+        <description isPrivate="false">4th street</description>
     </persons>
 </addressbook>
```
Comment (remo5000) [inline src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml]: this shouldnt be working

File: src/test/data/XmlSerializableAddressBookTest/typicalPersonsAddressBook.xml [hunk 16]
```diff
@@ -1,50 +1,50 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<!-- TaskManager save file which contains the same Task values as in TypicalPersons#getTypicalAddressBook() -->
-<addressbook>
-    <persons>
+<!-- AddressBook save file which contains the same Person values as in TypicalPersons#getTypicalAddressBook() -->
```
Comment (remo5000) [inline src/test/data/XmlSerializableAddressBookTest/typicalPersonsAddressBook.xml]: comment

File: src/test/java/seedu/address/commons/util/XmlUtilTest.java [hunk 17]
```diff
@@ -28,19 +28,19 @@
...
+    private static final Path INVALID_TASK_FIELD_FILE = TEST_DATA_FOLDER.resolve("invalidTaskField.xml");
+    private static final Path VALID_TASK_FILE = TEST_DATA_FOLDER.resolve("validTask.xml");
+    private static final Path TEMP_FILE = TestUtil.getFilePathInSandboxFolder("tempTaskManager.xml");
 
     private static final String INVALID_PHONE = "9482asf424";
 
     private static final String VALID_NAME = "Hans Muster";
     private static final String VALID_PHONE = "9482424";
     private static final String VALID_EMAIL = "hans@example";
     private static final String VALID_ADDRESS = "4th street";
```
Comment (remo5000) [inline src/test/java/seedu/address/commons/util/XmlUtilTest.java]: update names to new ones

File: src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java [hunk 18]
```diff
@@ -19,92 +19,94 @@
...
-                new XmlAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, INVALID_ADDRESS, VALID_TAGS);
+        XmlAdaptedTask task =
+                new XmlAdaptedTask(VALID_NAME, VALID_DUE_DATE, VALID_PRIORITY_VALUE, INVALID_DESCRIPTION, VALID_LABELS);
         String expectedMessage = Description.MESSAGE_DESCRIPTION_CONSTRAINTS;
-        Assert.assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
+        Assert.assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
     }
 
     @Test
     public void toModelType_nullAddress_throwsIllegalValueException() {
```
Comment (remo5000) [inline src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java]: update to description

File: src/test/java/seedu/address/storage/XmlAddressBookStorageTest.java [hunk 19]
```diff
@@ -47,80 +47,80 @@ private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
...
          * That means you should not have more than one exception test in one method
          */
     }
 
     @Test
-    public void readAddressBook_invalidPersonAddressBook_throwDataConversionException() throws Exception {
+    public void readTaskManager_invalidTaskTaskManager_throwDataConversionException() throws Exception {
         thrown.expect(DataConversionException.class);
-        readAddressBook("invalidPersonAddressBook.xml");
+        readTaskManager("invalidPersonAddressBook.xml");
```
Comment (remo5000) [inline src/test/java/seedu/address/storage/XmlAddressBookStorageTest.java]: - [ ] rename XMLs after merging in UI

File: src/main/java/seedu/address/storage/XmlAdaptedLabel.java [hunk 20]
```diff
@@ -8,44 +8,43 @@
...
-        tagName = source.labelName;
+    public XmlAdaptedLabel(Label source) {
+        labelName = source.labelName;
     }
 
     /**
-     * Converts this jaxb-friendly adapted tag object into the model's Label object.
+     * Converts this jaxb-friendly adapted label object into the model's Label object.
      *
      * @throws IllegalValueException if there were any data constraints violated in the adapted person
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlAdaptedLabel.java]: comment

File: src/main/java/seedu/address/storage/XmlAdaptedTask.java [hunk 21]
```diff
@@ -20,66 +20,66 @@
...
+     * Constructs an XmlAdaptedTask.
      * This is the no-arg constructor that is required by JAXB.
      */
-    public XmlAdaptedPerson() {
-    }
+    public XmlAdaptedTask() {}
 
     /**
-     * Constructs an {@code XmlAdaptedPerson} with the given person details.
+     * Constructs an {@code XmlAdaptedTask} with the given person details.
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlAdaptedTask.java]: comment

File: src/main/java/seedu/address/storage/XmlAdaptedTask.java [hunk 22]
```diff
@@ -20,66 +20,66 @@
...
+        labelled = source.getLabels().stream()
+                .map(XmlAdaptedLabel::new)
                 .collect(Collectors.toList());
     }
 
     /**
-     * Converts this jaxb-friendly adapted person object into the model's Task object.
+     * Converts this jaxb-friendly adapted task object into the model's Task object.
      *
      * @throws IllegalValueException if there were any data constraints violated in the adapted person
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlAdaptedTask.java]: comment

File: src/main/java/seedu/address/storage/XmlSerializableAddressBook.java [hunk 23]
```diff
@@ -21,22 +21,22 @@
     public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
```
Comment (remo5000) [inline src/main/java/seedu/address/storage/XmlSerializableAddressBook.java]: local variable

File: src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml [hunk 24]
```diff
@@ -3,15 +3,15 @@
     <!-- Valid Task -->
     <persons>
```
Comment (remo5000) [inline src/test/data/XmlAddressBookStorageTest/invalidAndValidPersonAddressBook.xml]: update tags

File: src/test/data/XmlSerializableAddressBookTest/duplicatePersonAddressBook.xml [hunk 25]
```diff
@@ -1,20 +1,20 @@
...
+        <dueDate>94351253</dueDate>
+        <priorityValue>alice@example.com</priorityValue>
+        <description>123, Jurong West Ave 6, #08-111</description>
         <tagged>friends</tagged>
-    </persons>
+    </tasks>
 
-    <!-- Task with same identity as Alice Pauline -->
-    <persons>
+    <!-- Person with same identity as Alice Pauline -->
```
Comment (remo5000) [inline src/test/data/XmlSerializableAddressBookTest/duplicatePersonAddressBook.xml]: comment

File: src/test/data/XmlSerializableAddressBookTest/invalidPersonAddressBook.xml [hunk 26]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<addressbook>
-    <!-- Task with invalid email field -->
-    <persons>
+<taskmanager>
+    <!-- Person with invalid priorityValue field -->
```
Comment (remo5000) [inline src/test/data/XmlSerializableAddressBookTest/invalidPersonAddressBook.xml]: comment

File: src/test/data/XmlUtilTest/invalidTaskField.xml [hunk 27]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!-- Person with an invalid dueDate field -->
```
Comment (remo5000) [inline src/test/data/XmlUtilTest/invalidTaskField.xml]: comment

File: src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java [hunk 28]
```diff
@@ -19,92 +19,94 @@
 
 public class XmlAdaptedPersonTest {
```
Comment (remo5000) [inline src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java]: - [ ] rename class after merging ui

File: src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java [hunk 29]
```diff
@@ -19,92 +19,94 @@
...
-            .map(XmlAdaptedTag::new)
+    private static final String VALID_DUE_DATE = BENSON.getDueDate().toString();
+    private static final String VALID_PRIORITY_VALUE = BENSON.getPriorityValue().toString();
+    private static final String VALID_DESCRIPTION = BENSON.getDescription().toString();
+    private static final List<XmlAdaptedLabel> VALID_LABELS = BENSON.getLabels().stream()
+            .map(XmlAdaptedLabel::new)
             .collect(Collectors.toList());
 
     @Test
     public void toModelType_validPersonDetails_returnsPerson() throws Exception {
```
Comment (remo5000) [inline src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java]: update to task

File: src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java [hunk 30]
```diff
@@ -19,92 +19,94 @@
...
-        XmlAdaptedPerson person = new XmlAdaptedPerson(BENSON);
+        XmlAdaptedTask person = new XmlAdaptedTask(BENSON);
```
Comment (remo5000) [inline src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java]: local variable

File: src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java [hunk 31]
```diff
@@ -19,92 +19,94 @@
...
+                VALID_DESCRIPTION, VALID_LABELS);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName());
         Assert.assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
     }
 
     @Test
     public void toModelType_invalidPhone_throwsIllegalValueException() {
-        XmlAdaptedPerson person =
-                new XmlAdaptedPerson(VALID_NAME, INVALID_PHONE, VALID_EMAIL, VALID_ADDRESS, VALID_TAGS);
+        XmlAdaptedTask person =
```
Comment (remo5000) [inline src/test/java/seedu/address/storage/XmlAdaptedPersonTest.java]: local variable

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 32]
```diff
@@ -21,13 +21,13 @@
...
-    private AddressBookStorage addressBookStorage;
+    private TaskManagerStorage addressBookStorage;
     private UserPrefsStorage userPrefsStorage;
 
 
-    public StorageManager(AddressBookStorage addressBookStorage, UserPrefsStorage userPrefsStorage) {
+    public StorageManager(TaskManagerStorage taskManagerStorage, UserPrefsStorage userPrefsStorage) {
         super();
-        this.addressBookStorage = addressBookStorage;
+        this.addressBookStorage = taskManagerStorage;
```
Comment (jiaxiu11) [inline src/main/java/seedu/address/storage/StorageManager.java]: local variable

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 33]
```diff
@@ -52,39 +52,39 @@ public void saveUserPrefs(UserPrefs userPrefs) throws IOException {
...
     @Override
-    public void saveAddressBook(ReadOnlyTaskManager addressBook) throws IOException {
-        saveAddressBook(addressBook, addressBookStorage.getAddressBookFilePath());
+    public void saveTaskManager(ReadOnlyTaskManager taskManager) throws IOException {
+        saveTaskManager(taskManager, addressBookStorage.getTaskManagerFilePath());
     }
 
     @Override
-    public void saveAddressBook(ReadOnlyTaskManager addressBook, Path filePath) throws IOException {
+    public void saveTaskManager(ReadOnlyTaskManager addressBook, Path filePath) throws IOException {
```
Comment (jiaxiu11) [inline src/main/java/seedu/address/storage/StorageManager.java]: parameter

File: src/main/java/seedu/address/storage/XmlFileStorage.java [hunk 34]
```diff
@@ -27,10 +27,10 @@ public static void saveDataToFile(Path file, XmlSerializableAddressBook addressB
     /**
      * Returns task manager in the file or an empty task manager
      */
-    public static XmlSerializableAddressBook loadDataFromSaveFile(Path file) throws DataConversionException,
-            FileNotFoundException {
+    public static XmlSerializableTaskManager loadDataFromSaveFile(Path file) throws DataConversionException,
+                                                                            FileNotFoundException {
```
Comment (jiaxiu11) [inline src/main/java/seedu/address/storage/XmlFileStorage.java]: unnecessary whitespace?

File: src/test/java/seedu/address/ui/TaskListPanelTest.java [hunk 35]
```diff
@@ -93,16 +93,16 @@ public void performanceTest() throws Exception {
...
-            builder.append("</persons>\n");
+            builder.append("<dueDate>000</dueDate>\n");
+            builder.append("<priorityValue>a@aa</priorityValue>\n");
+            builder.append("<description>a</description>\n");
+            builder.append("</tasks>\n");
         }
-        builder.append("</addressbook>\n");
+        builder.append("</taskmanager>\n");
 
         Path manyPersonsFile = Paths.get(TEST_DATA_FOLDER + "manyPersons.xml");
```
Comment (jiaxiu11) [inline src/test/java/seedu/address/ui/TaskListPanelTest.java]: local variable
